src/plstring.c

```diff
@@ -0,0 +1,540 @@
+        for (power = POWERS_OF_TEN[len - 1]; power > 0; power /= 10) {
+            digit     = (u16) (num / power);
+            remainder = (u32) (num - (power * digit));
+
+            if (paddingMode == NUMBER_PADDING_MODE_ZEROES)      // this branch is hit if the user specified this padding mode, or any time after we hit a non-zero digit
+            {
+                dst->buffer[dst->len++] = (digit < 10) ? charMap[digit] : 0xe2;
+            }
+            else if (digit != 0 || power == 1)                  // once we hit this branch, we should always hit the above branch on future digits
+            {
+                paddingMode = NUMBER_PADDING_MODE_ZEROES;
+                dst->buffer[dst->len++] = (digit < 10) ? charMap[digit] : 0xe2;
+            }
+            else if (paddingMode == NUMBER_PADDING_MODE_SPACES) // not in zero-padding mode, and found a leading zero, pad out with the appropriate width space
+            {
+                dst->buffer[dst->len++] = (charWidth == CHARACTER_WIDTH_FULL) ? 0x1 : 0x1e2;
+            }
+
+            num = remainder;
+        }
+
+        dst->buffer[dst->len] = TERMINATOR;
+        return;
+    }
+
+    GF_ASSERT(0);
+}
+
+void PLString_FromInt64 (PLString * string, u64 number, u32 len, NUMBER_PADDING_MODE paddingMode, CHARACTER_WIDTH charWidth)
+{
+    static const u64 POWERS_OF_TEN[] = {
+        1,
+        10,
+        100,
+        1000,
+        10000,
+        100000,
+        1000000,
+        10000000,
+        100000000,
+        1000000000,
+        10000000000,
+        100000000000,
+        1000000000000,
+        10000000000000,
+        100000000000000,
+        1000000000000000,
+        10000000000000000,
+        100000000000000000,
+        1000000000000000000,
+        10000000000000000000
+    };
+    static const u16 FULL_WIDTH_DIGITS[] = {
+        0xa2,   // 0
+        0xa3,   // 1
+        0xa4,   // 2
+        0xa5,   // 3
+        0xa6,   // 4
+        0xa7,   // 5
+        0xa8,   // 6
+        0xa9,   // 7
+        0xaa,   // 8
+        0xab    // 9
+    };
+    static const u16 HALF_WIDTH_DIGITS[] = {
+        0x121,  // 0
+        0x122,  // 1
+        0x123,  // 2
+        0x124,  // 3
+        0x125,  // 4
+        0x126,  // 5
+        0x127,  // 6
+        0x128,  // 7
+        0x129,  // 8
+        0x12a   // 9
+    };
+
+    PLSTRING_CHECK(string);
+
+    // make sure the number will fit
+    if (string->size > len) {
+        u64 remainder, digit, power;
+        const u16 * charMap = (charWidth == CHARACTER_WIDTH_FULL) ? FULL_WIDTH_DIGITS : HALF_WIDTH_DIGITS;
+
+        PLString_Clear(string);
+
+        // curse forwards through the integer digit by digit, appending individual digits along the way
+        for (power = POWERS_OF_TEN[len - 1]; power > 0; power /= 10) {
+            digit     = (u16) (number / power);
+            remainder = (u64) (number - (power * digit));
```

Comment:
Do not align assignments. There should be no space between `(cast)` and `(castee)`.